internal/distro/rhel85/distro.go

```diff
@@ -252,8 +266,38 @@ func (t *imageType) PackageSets(bp blueprint.Blueprint) map[string]rpmmd.Package
 
 	// package sets from flags
 	if t.bootable {
-		// add boot sets
-		mergedSets[osPkgsKey] = mergedSets[osPkgsKey].Append(archSets[bootPkgsKey]).Append(distroSets[bootPkgsKey])
+		var addLegacyBootPkg bool
+		var addUEFIBootPkg bool
+
+		switch bt := t.getBootType(); bt {
+		case LegacyBootType:
+			addLegacyBootPkg = true
+		case UEFIBootType:
+			addUEFIBootPkg = true
+		case HybridBootType:
+			addLegacyBootPkg = true
+			addUEFIBootPkg = true
+		default:
+			panic(fmt.Sprintf("unsupported boot type: %q", bt))
+		}
+
+		if addLegacyBootPkg {
+			if set, exists := distroSets[bootLegacyPkgsKey]; exists {
```

Comment:
I think this check is unnecessary because, if it doesn't exist, the `Append()` method gets an empty list and does nothing.  it's not an issue, just might be simpler to read without the check.

Reply:
Thanks. I simplified it!